azurelinuxagent/common/cgroup.py

```diff
@@ -13,104 +13,139 @@
+    def __init__(self, extension_name, cgroup_path, controller_type):
+        self.name = extension_name
+        self.path = cgroup_path
+        self.controller = controller_type
 
+    def _get_cgroup_file(self, file_name):
+        return os.path.join(self.path, file_name)
 
-    def get_file_contents(self, hierarchy, file_name):
+    def get_file_contents(self, controller, file_name):
         """
-        Retrieve the value of a parameter from a hierarchy.
+        Retrieve the value of a parameter from a controller.
 
-        :param str hierarchy: Name of cgroup metric hierarchy
-        :param str file_name: Name of file within that metric hierarchy
+        :param str controller: Name of cgroup metric controller
+        :param str file_name: Name of file within that metric controller
         :return: Entire contents of the file
         :rtype: str
         """
-        if hierarchy in self.cgroups:
-            parameter_file = self._get_cgroup_file(hierarchy, file_name)
+        if controller == self.controller:
+            parameter_file = self._get_cgroup_file(file_name)
 
             try:
                 return fileutil.read_file(parameter_file)
             except Exception:
-                raise CGroupsException("Could not retrieve cgroup file {0}/{1}".format(hierarchy, file_name))
+                raise CGroupsException("Could not retrieve cgroup file {0}/{1}".format(self.path, file_name))
         else:
-            raise CGroupsException("{0} subsystem not available in cgroup {1}. cgroup paths: {2}".format(
-                hierarchy, self.name, self.cgroups))
+            raise CGroupsException("Incorrect controller: {0} passed for cgroup: {1}. Cgroup path is: {2}".format(
+                controller, self.name, self.path))
 
-    def get_parameter(self, hierarchy, parameter_name):
+    def get_parameter(self, controller, parameter_name):
         """
         Retrieve the value of a parameter from a hierarchy.
         Assumes the parameter is the sole line of the file.
 
-        :param str hierarchy: Name of cgroup metric hierarchy
+        :param str controller: Name of cgroup metric hierarchy
         :param str parameter_name: Name of file within that metric hierarchy
         :return: The first line of the file, without line terminator
         :rtype: str
         """
         result = ""
         try:
-            values = self.get_file_contents(hierarchy, parameter_name).splitlines()
+            values = self.get_file_contents(controller, parameter_name).splitlines()
             result = values[0]
         except IndexError:
-            parameter_filename = self._get_cgroup_file(hierarchy, parameter_name)
+            parameter_filename = self._get_cgroup_file(parameter_name)
+            logger.error("File {0} is empty but should not be".format(parameter_filename))
+        except CGroupsException:
+            # ignore if the file does not exist yet
```

Comment:
this comment is misleading, or we are catching the wrong exception... seems to me we can get CGroupsException under conditions other that the file not existing

Reply:
Returning None here and handling it upstream.